### ISSUE TITLE
Version conversions lead to erroneous deprecation warnings

### ISSUE THREAD

**Issue by howardjohn** [kubernetes/kubernetes#94981]
**What happened**:
On an empty cluster:

```
$ kubectl get ingress
Warning: extensions/v1beta1 Ingress is deprecated in v1.14+, unavailable in v1.22+; use networking.k8s.io/v1 Ingress
No resources found in default namespace.
```

**What you expected to happen**:

No warnings if there is no corrective action. I don't even have any Ingress in my cluster, so telling me to use `networking.k8s.io/v1` isn't very helpful.

If I explicitly do `k get ingresses.networking.k8s.io` there is no warning. I would expect this to be the default if we are going to warn otherwise?

**How to reproduce it (as minimally and precisely as possible)**:

Spin up a cluster, run `kubectl get ingress`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.0-rc.2", GitCommit:"27bb2a4a0a5cb8330178d19e57fa61fffa895c98", GitTreeState:"clean", BuildDate:"2020-07-21T17:39:35Z", GoVersion:"go1.14.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-14T07:30:52Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: kind

**Comment by fedebongio** [comment]
/assign @liggitt 
Jordan trying with you first, let me know if you need our help on this one.

**Comment by liggitt** [comment]
This is fixed in 1.20 in https://github.com/kubernetes/kubernetes/pull/94309. Because this missed 1.19, a cherry-pick to change API group priority doesn't seem to meet the bar for cherrypicks for 1.19.

For ambiguous requests like `kubectl get ingress`, `extensions/v1beta1` has historically been prioritized for backwards compatibility. I agree that with the increased visibility to use of deprecated versions, and since ingress is the only resource remaining in extensions/v1beta1, we should begin preferring networking.k8s.io/v1, and will in 1.20.

/close
